### ISSUE TITLE
Can not set stylesheet path from `c` via the `c-api`

### ISSUE THREAD

**Issue by michabay05** [linebender/resvg#870]
### Context
Most of the important `usvg` options written in `rust` are available in the `c` api. Some of the accessible options are setting the language, resource directory, rendering modes, and so on. However, the stylesheet option, which is available in the `rust` api, is *not* available in the `c` api.

https://github.com/linebender/resvg/blob/bda7db0d0b2eb32c5f63f90279cec19613945a9d/crates/usvg/src/main.rs#L28-L29
------
### Question
- Are there plans to add this option to the `c-api`? Something like this perhaps?
```h
/**
 *
 * @brief Sets a path that will be used when resolving CSS attributes.
 *
 */
void resvg_options_set_stylesheet(resvg_options *opt, const char *path);
```

**P.S.**: Great work on the library!

**Comment by LaurenzV** [comment]
Yeah, this could definitely be added. Feel free to open a PR, although I'm not sure I would be able to review it because my C knowledge isn't great, but I'm sure someone else would be able to.

**Comment by michabay05** [comment]
I'm interested in opening a PR to add this feature (here's the PR #873). In trying to do so, I found something to be a bit confusing. As you can see, the command line argument below requests a path to be provided to inject a stylesheet. 

https://github.com/linebender/resvg/blob/bda7db0d0b2eb32c5f63f90279cec19613945a9d/crates/usvg/src/main.rs#L28-L29

So, I compiled `resvg` into a binary, provided the path to the stylesheet, and ran the command. Unsuprisingly, it worked great.
```console
$ target/debug/resvg --stylesheet test.css some.svg some.png
```
However, I came across some issues when trying to replicate those results when using `resvg` as a library. I started from the miminal example (provided here: [`resvg/examples/minimal.rs`](https://github.com/michabay05/resvg/blob/main/crates/resvg/examples/minimal.rs)). For some reason, it didn't work as intended. After some digging, it turned out that `resvg` as a binary expects a path to the stylesheet; however, `resvg` as a library expects the content of the stylesheet provided as a string. For more detail about the inconsistency, take a look a the dropdown.

<details>
<summary><strong>More detail</strong></summary>
<p>
When running `resvg` as a binary, the path provided from the command line is treated as a `PathBuf` and then, the stylesheet's content is read.
https://github.com/linebender/resvg/blob/bda7db0d0b2eb32c5f63f90279cec19613945a9d/crates/resvg/src/main.rs#L242
https://github.com/linebender/resvg/blob/bda7db0d0b2eb32c5f63f90279cec19613945a9d/crates/resvg/src/main.rs#L554-L562
On the other hand, the `usvg` parser, used in the minimal example, treats the stylesheet as a `String`. From that point on, it assumes `stylesheet` contains the contents of the stylesheet.
https://github.com/linebender/resvg/blob/6df63519f04580bd74e31b25e9ee82d4ace145da/crates/usvg/src/parser/options.rs#L98-L100
</p>
</details> 

I anticipate the difference when using `resvg` as a library and binary would cause confusion. In my opinion, the best course of action would be to treat the stylesheet as a `PathBuf` throughout the entire codebase. In addition to establishing consistency, when using the `c-api`, the additional layer of issues associated with managing memory to read from the file and save to a string would be eliminated.

How would like to handle this issue? I'm willing to open to creating another issue and creating a PR for it, but I wanted to get your opinion on it before I do that.

**Comment by LaurenzV** [comment]
This is intentional, resvg as a library should be pure and not have any interaction with the filesystem whatsoever, so I'm afraid we can't change that.

Why can't we just ask the user of the C api to just load the file from their side and then pass the contents to resvg?

**Comment by michabay05** [comment]
Understood. Security and portability are important considerations. I've updated my PR for this issue (#873) to account for this fact.
